rdflib/term.py

```diff
@@ -662,6 +670,10 @@ def normalize(self) -> "Literal":
         else:
             return self
 
+    @property
+    def ill_formed(self) -> bool:
+        return self._ill_formed
+
```

Comment:
It may be worth considering to make this a three valued property:

```suggestion
    @property
    def ill_formed(self) -> Optional[bool]:
        return self._ill_formed
```

And only return True/False if it can be established conclusively, the current approach will yield `ill_formed` for values which may actually be well formed but just for a datatype that is not registered.

Reply:
Also, should this not rather be "ill_typed" instead of ill_formed?

https://www.w3.org/TR/rdf11-concepts/#section-Graph-Literal

> Otherwise, the literal is ill-typed and no literal value can be associated with the literal. Such a case produces a semantic inconsistency but is not syntactically ill-formed. Implementations must accept ill-typed literals and produce RDF graphs from them. Implementations may produce warnings when encountering ill-typed literals.